src/Coin.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Coin.h"
+
+#include "Bitcoin/Address.h"
+#include "Bitcoin/CashAddress.h"
+#include "Bitcoin/Bech32Address.h"
+#include "Ethereum/Address.h"
+#include "Icon/Address.h"
+#include "Nimiq/Address.h"
+#include "Ripple/Address.h"
+#include "Tendermint/Address.h"
+#include "Tezos/Address.h"
+#include "Zcash/TAddress.h"
+
+#include <TrustWalletCore/TWHRP.h>
+#include <TrustWalletCore/TWP2PKHPrefix.h>
+#include <TrustWalletCore/TWP2SHPrefix.h>
+
+using namespace TW;
+
+std::string TW::loadAddress(TWCoinType coin, const Data& data) {
+    switch (coin) {
+    case TWCoinTypeBinance:
+        return Tendermint::Address(HRP_BINANCE, data).string();
+
+    case TWCoinTypeBitcoin:
+        return Bitcoin::Bech32Address(HRP_BITCOIN, 0, data).string();
+
+    case TWCoinTypeBitcoinCash:
+        return Bitcoin::CashAddress(data).string();
+
+    case TWCoinTypeCallisto:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypePoa:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+        return Ethereum::Address(data).string();
+
+    case TWCoinTypeICON:
+        return Icon::Address(data, TWIconAddressTypeAddress).string();
+
+    case TWCoinTypeLitecoin:
+        return Bitcoin::Bech32Address(HRP_LITECOIN, 0, data).string();
+
+    case TWCoinTypeNimiq:
+        return Nimiq::Address(data).string();
+
+    case TWCoinTypeRipple:
+        return Ripple::Address(data).string();
+
+    case TWCoinTypeDash:
+    case TWCoinTypeTron:
```

Comment:
No TronAddress? I remember Tron uses sha3 instead of sha256 for keyhash

Reply:
We are missing a Tron::Address class...

Reply:
Ok, There is a `TronAddress.swift`, that's why iOS still works